sympy/polys/domains/modularinteger.py

```diff
@@ -136,15 +136,7 @@ def __pow__(self, exp):
         else:
             val = self.val
 
-        m = self.mod
-        r = self.dom.one
-        while exp > 0:
-            if exp % 2:
-                r = (r * val) % m
-            val = (val * val) % m
-            exp //= 2
-
-        return self.__class__(r)
+        return self.__class__(pow(int(val), exp, self.mod))
```

Comment:
Is it necessary to convert `val` to an `int`?

Reply:
I think it would just recursively call this method otherwise.

Reply:
`gmp` has also implemented the three argument [`mpz_powm`](https://gmplib.org/manual/Integer-Exponentiation.html#Integer-Exponentiation). I would expect that `gmpy` would be using that.

Reply:
I see. `val` is the underlying integer. Except if exp is negative it is `self.invert()`, but that will recursively call this (it could be `self.invert().val` to save a call). So removing `int` should be preferred.